src/InputToolbar.tsx

```diff
@@ -33,142 +27,83 @@ const styles = StyleSheet.create({
   },
 })
 
-export interface InputToolbarProps {
+export interface InputToolbarProps<TMessage extends IMessage> {
   options?: { [key: string]: any }
   optionTintColor?: string
   containerStyle?: StyleProp<ViewStyle>
   primaryStyle?: StyleProp<ViewStyle>
   accessoryStyle?: StyleProp<ViewStyle>
-  renderAccessory?(props: InputToolbarProps): React.ReactNode
-  renderActions?(props: Actions['props']): React.ReactNode
-  renderSend?(props: Send['props']): React.ReactNode
-  renderComposer?(props: Composer['props']): React.ReactNode
+  renderAccessory?(props: InputToolbarProps<TMessage>): React.ReactNode
+  renderActions?(props: ActionsProps): React.ReactNode
+  renderSend?(props: SendProps<TMessage>): React.ReactNode
+  renderComposer?(props: ComposerProps): React.ReactNode
   onPressActionButton?(): void
 }
 
-export default class InputToolbar extends React.Component<
-  InputToolbarProps,
-  { position: string }
-> {
-  static defaultProps = {
-    renderAccessory: null,
-    renderActions: null,
-    renderSend: null,
-    renderComposer: null,
-    containerStyle: {},
-    primaryStyle: {},
-    accessoryStyle: {},
-    onPressActionButton: () => {},
-  }
-
-  static propTypes = {
-    renderAccessory: PropTypes.func,
-    renderActions: PropTypes.func,
-    renderSend: PropTypes.func,
-    renderComposer: PropTypes.func,
-    onPressActionButton: PropTypes.func,
-    containerStyle: StylePropType,
-    primaryStyle: StylePropType,
-    accessoryStyle: StylePropType,
-  }
-
-  state = {
-    position: 'absolute',
-  }
-
-  keyboardWillShowListener?: EmitterSubscription = undefined
-  keyboardWillHideListener?: EmitterSubscription = undefined
-
-  componentDidMount() {
-    this.keyboardWillShowListener = Keyboard.addListener(
+export function InputToolbar<TMessage extends IMessage = IMessage>(
+  props: InputToolbarProps<TMessage>,
+) {
+  const [position, setPosition] = useState('absolute')
+  useEffect(() => {
+    const keyboardWillShowListener = Keyboard.addListener(
```

Comment:
I haven't test it in gifted chat. But we faced a lot of issues with keyboard, while working on stream-chat [RN sdk](https://github.com/GetStream/stream-chat-react-native). One of the issues was as following:
- open the keyboard
- switch to some other app
- dismiss the keyboard
- come back to chat app

At this point, keyboard has dismissed in chat application as well (I guess because keyboard events work on native level), but js thread couldn't act on it because app was in background. So it resulted in white-patch at a place where keyboard was supposed to be.

Our solution was to dismiss the keyboard and remove the listeners, when app went to background.

But again, I haven't tested with gifted chat, so maybe its not an issue here. But something to test, if not done :)